src/rust/aggregator/src/pipelines/coins.rs

```diff
@@ -0,0 +1,138 @@
+        )
+        .execute(&self.pool)
+        .await
+        .map_err(|e| PipelineError::ProcessingError(anyhow!(e)))?;
+        Ok(())
+    }
+
+    fn poll_interval(&self) -> Option<std::time::Duration> {
+        Some(std::time::Duration::from_secs(60 * 60))
+    }
+
+    async fn process_and_save_internal(&mut self) -> PipelineAggregationResult {
+        let coins = sqlx::query!(
+            r#"
+                WITH market_coins AS (
+                    SELECT DISTINCT
+                        base_account_address AS address,
+                        base_module_name AS module,
+                        base_struct_name AS struct
+                    FROM
+                        market_registration_events
+                    WHERE base_account_address IS NOT NULL
+                    AND base_module_name IS NOT NULL
+                    AND base_struct_name IS NOT NULL
+                    UNION
+                    SELECT DISTINCT
+                        quote_account_address AS address,
+                        quote_module_name AS module,
+                        quote_struct_name AS struct
+                    FROM
+                        market_registration_events
+                )
+                SELECT
+                    *
+                FROM
+                    market_coins AS m
+                WHERE
+                    NOT EXISTS (
+                        SELECT
+                            *
+                        FROM
+                            aggregator.coins AS c
+                        WHERE
+                            c.address = m.address
+                        AND
+                            c.module = m.module
+                        AND
+                            c.struct = m.struct
+                    )
+            "#,
+        )
+        .fetch_all(&self.pool)
+        .await
+        .map_err(|e| PipelineError::ProcessingError(anyhow!(e)))?;
+        for coin in coins {
+            let (address, module, struct_name) = (coin.address.unwrap(), coin.module.unwrap(), coin.r#struct.unwrap());
+            let move_struct_tag_str = format!("{}::{}::{}", address, module, struct_name);
+            let coin_info = get_coin_info(move_struct_tag_str).await;
+            match coin_info {
+                Ok(coin_info) => {
+                    let CoinInfo { name, symbol, decimals } = coin_info;
+                    sqlx::query!(
+                        "INSERT INTO aggregator.coins VALUES ($1, $2, $3, $4, $5, $6)",
+                        name, symbol, decimals, address, module, struct_name
+                    )
+                    .execute(&self.pool)
+                    .await
+                    .map_err(|e| PipelineError::ProcessingError(anyhow!(e)))?;
+                },
+                Err(GetCoinInfoError::RateLimited) => {},
+                Err(GetCoinInfoError::Other(e)) => {return Err(PipelineError::ProcessingError(anyhow!(e)));},
+            }
+        }
+        Ok(())
```

Comment:
We might want to have a table that tracks `aggregator.coins_last_time_rate_limited`

Then we have a constant like `TIME_TO_WAIT_AFTER_GETTING_RATE_LIMITED`

If it hasn't been this long since the last query, do not try querying at all

Once this is implemented, we can run the pipeline on a 1s polling interval to give real-time market registration updates